File: leetcode-104.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    ans = 0
    def maxDepth(self, root: TreeNode) -> int:
        def dfs(u, step=1):
            if u is None:
                return
            self.ans = max(self.ans, step)
            dfs(u.left, step+1)
            dfs(u.right, step+1)
        dfs(root)
        return self.ans

File: leetcode/leetcode-104.cpp
/*
 *@File Name          :leetcode-104.cpp
 *@Author             :cc
 *@Version            :1.0.0
 *@Date               :2022/3/27 9:26
 *@Description        :
 *@Function List      :
 *@History            :
 */


struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;

    TreeNode() : val(0), left(nullptr), right(nullptr) {}

    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}

    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int maxDepth(TreeNode *root) {
        if (root == nullptr) return 0;
        return 1 + max(maxDepth(root->left), maxDepth(root->right));
    }
};